RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/PixelClustering.h

```diff
@@ -164,25 +172,39 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE::pixelClustering {
     static constexpr uint32_t maxElementsPerBlock =
         cms::alpakatools::round_up_by(TrackerTraits::maxPixInModule / maxIterGPU, 128);
 
+    // FIXME: this is just an estimate, to be studied and optimised
+    //static constexpr uint32_t maxFakesInModule = TrackerTraits::maxPixInModule * 2 / 5;
+
     ALPAKA_FN_ACC void operator()(Acc1D const& acc,
                                   SiPixelDigisSoAView digi_view,
+                                  SiPixelDigisSoAView fakes_view,
+                                  bool applyDigiMorphing,
+                                  uint32_t* morphingModules,
+                                  uint32_t nMorphingModules,
+                                  uint32_t maxFakesInModule,
                                   SiPixelClustersSoAView clus_view,
                                   const unsigned int numElements) const {
       static_assert(TrackerTraits::numberOfModules < ::pixelClustering::maxNumModules);
 
       auto& lastPixel = alpaka::declareSharedVar<unsigned int, __COUNTER__>(acc);
+      auto& fakePixels = alpaka::declareSharedVar<unsigned int, __COUNTER__>(acc);
 #ifdef GPU_DEBUG
       auto& goodPixels = alpaka::declareSharedVar<uint32_t, __COUNTER__>(acc);
 #endif
 
       const uint32_t lastModule = clus_view[0].moduleStart();
       for (uint32_t module : cms::alpakatools::independent_groups(acc, lastModule)) {
+        auto block = alpaka::getIdx<alpaka::Grid, alpaka::Blocks>(acc)[0u];
+
         auto firstPixel = clus_view[1 + module].moduleStart();
         uint32_t thisModuleId = digi_view[firstPixel].moduleId();
+        uint32_t rawModuleId = digi_view[firstPixel].rawIdArr();
+        applyDigiMorphing =
+            applyDigiMorphing && pixelStatus::isMorphingModule(rawModuleId, morphingModules, nMorphingModules);
```

Comment:
I think this is wrong: a kernel block may loop over different modules (line 196); once it encounters a module that is not in the list, it will change the `applyDigiMorphing` to `false`; at that point all other modules will be ignored.